android-components/components/feature/media/src/main/java/mozilla/components/feature/media/fullscreen/MediaSessionFullscreenFeature.kt

```diff
@@ -35,51 +35,53 @@ class MediaSessionFullscreenFeature(
             flow.map {
                 it.tabs + it.customTabs
             }.map { tab ->
-                tab.filter { it.mediaSessionState != null && it.mediaSessionState!!.fullscreen }
-            }.distinctUntilChanged().collect { states ->
-                processFullscreen(states)
-                processDeviceSleepMode(states)
+                tab.firstOrNull { it.mediaSessionState != null && it.mediaSessionState!!.fullscreen }
+            }.distinctUntilChanged { old, new ->
+                old?.mediaSessionState?.fullscreen == new?.mediaSessionState?.fullscreen &&
+                    old?.mediaSessionState?.playbackState == new?.mediaSessionState?.playbackState &&
+                    old?.content?.pictureInPictureEnabled == new?.content?.pictureInPictureEnabled &&
+                    old?.mediaSessionState?.elementMetadata == new?.mediaSessionState?.elementMetadata
+            }.collect { state -> /* there should only be one fullscreen session */
```

Comment:
Is there anything to your knowledge that enforces the constraint mentioned in this comment? I'm not familiar with this domain space

Reply:
I'd say it's mainly because it is not technically possible to have two tabs in fullscreen at the same time. Does it seem to make sense to you?